pkg/controller/mcmhub/gitrepo_sync.go

```diff
@@ -116,19 +118,79 @@ func (r *ReconcileSubscription) UpdateGitDeployablesAnnotation(sub *appv1.Subscr
 		// Check and add cluster-admin annotation for multi-namepsace application
 		r.AddClusterAdminAnnotation(sub)
 
-		updated = true
-
 		if annotations[appv1.AnnotationDeployables] == "" {
 			// this might have failed previously. Try again.
 			r.updateGitSubDeployablesAnnotation(sub)
+		}
 
+		if ifUpdateGitSubscriptionAnnotation(origsub, sub) {
 			updated = true
 		}
 	}
 
 	return updated, nil
 }
 
+// ifUpdateGitSubscriptionAnnotation compare given annoations between the two subscriptions. return true if no the same
+func ifUpdateGitSubscriptionAnnotation(origsub, newsub *appv1.Subscription) bool {
+	origanno := origsub.GetAnnotations()
+	newanno := newsub.GetAnnotations()
+
+	// 1. compare deployables list annoation
+	origdplmap := make(map[string]bool)
+
+	if origanno != nil {
+		dpls := origanno[appv1.AnnotationDeployables]
```

Comment:
@xiangjingli I feel we should check up the AnnotationTopo as well since we are intent to retire AnnotationDeployables in the future.